README.rst

```diff
@@ -99,21 +146,181 @@ Options
                             The .pypirc config file to use
-      --skip-existing       Continue uploading files if one already exists
+      --skip-existing       Continue uploading files if one already exists. (Only
+                            valid when uploading to PyPI. Other implementations
+                            may not support this.)
+      --cert path           Path to alternate CA bundle
+      --client-cert path    Path to SSL client certificate, a single file
+                            containing the private key and the certificate in PEM
+                            format
+
+
+API
+---
+
+``twine`` is written in Python.  Of course.  So if you work on a
+Python tool that uses ``twine`` for uploading or registering, you can
+import it.
+
+Note that reading environment variables, for example
+``TWINE_REPOSITORY``, is a feature of the command line tool.  the API
+does not try to read this.
+
+
+API package
+~~~~~~~~~~~
+
+Create a package object:
+
+.. code-block:: python
+
+    from twine.package import PackageFile
+    package = PackageFile.from_filename(filename, comment)
+
+Here ``filename`` is the name of a distribution (usually a wheel or a
+source distribution) and comment is a comment to include with the
+distribution file.  The comment may be ``None``.
+
+You can add a gpg signature:
+
+.. code-block:: python
+
+    package.add_gpg_signature(signature_filepath, signature_filename)
+
+You can sign a package:
+
+.. code-block:: python
+
+    package.sign(sign_with, identity)
+
+
+API repository
+~~~~~~~~~~~~~~
+
+Define a repository:
+
+.. code-block:: python
+
+    from twine.repository import Repository
+    repository = Repository(config["repository"], username, password)
+    repository.set_certificate_authority(ca_cert)
+    repository.set_client_certificate(client_cert)
+
+
+Register a package:
+
+.. code-block:: python
+
+    package = PackageFile.from_filename(filename, comment)
+    response = repository.register()
+
+Upload a package:
+
+.. code-block:: python
+
+    if repository.package_is_uploaded(package):
+        ...
+    response = repository.upload(package)
+    repository.close()
+
+
+API exceptions
+~~~~~~~~~~~~~~
+
+When things go wrong:
+
+.. code-block:: python
+
+    # A redirect was detected that the user needs to resolve.
+    from twine.exceptions import RedirectDetected
+
+    # A package file was provided that could not be found on the file system.
+    from twine.exceptions import PackageNotFound
+
+
+API register
+~~~~~~~~~~~~
+
+Since version 2.0 you can use this:
+
+.. code-block:: python
+
+    from twine.commands.register import register
+    upload('dist/mypkg-0.1-py2.py3-none-any.whl')
+
+It takes as only required argument the name of a file from which we
+read the package metadata.  This is usually a wheel or a source
+distribution.
+
+You can pass several optional keyword arguments.  This is the list,
+including the default values:
+
+.. code-block:: python
+
+   repository="pypi",
```

Comment:
1. We're not going to document the functions in `twine.commands`
2. Even if we were, you have inconsistent indentation here
3. We would document each parameter individually and explain the purpose, not just throw the defaults in like this.